packages/react-components/react-motion/library/src/factories/createPresenceComponent.ts

```diff
@@ -146,26 +152,30 @@ export function createPresenceComponent<MotionParams extends Record<string, Moti
           const atoms = visible ? presenceMotion.enter : presenceMotion.exit;
 
           const direction: PresenceDirection = visible ? 'enter' : 'exit';
-          const forceFinishMotion = !visible && isFirstMount;
+          const applyInitialStyles = !visible && isFirstMount;
+          const skipAnimation = optionsRef.current.skipMotions;
 
-          if (!forceFinishMotion) {
+          if (!applyInitialStyles) {
             handleMotionStart(direction);
           }
 
           const handle = animateAtoms(element, atoms, { isReducedMotion: isReducedMotion() });
+          handle.setMotionEndCallbacks(
+            () => handleMotionFinish(direction),
+            () => handleMotionCancel(direction),
+          );
```

Comment:
Will callbacks be called for the initial mount when `applyInitialStyles` is passed? It was not the thing before

Reply:
```ts
          if (applyInitialStyles) {
              // Heads up!
              // .finish() is used in this case to skip animation and apply animation styles immediately
            handle.finish();
            return;
          }

          handleRef.current = handle;
          handle.setMotionEndCallbacks(
            () => handleMotionFinish(direction),
            () => handleMotionCancel(direction),
          );

          if (skipAnimation) {
            handle.finish();
          }
```

this probably will be more accurate